odinochka.js

```diff
@@ -15,22 +15,55 @@ function newTabs(data) {
     data.tabs.forEach(o => chrome.tabs.create({url: o.url, pinned: o.pinned}))
 }
 
-function cssfilter(x) {
-    let node = document.getElementById("cssfilterstyle");
-    if(x.target.value != "") {
-        node.innerHTML = `a.tab:not([href*="${x.target.value}"]) {display:none} `;
-        // TODO someday when :has works, also hide the empty groups
-    }
-    else {
-        node.innerHTML = "";
+function debounce(func, wait, immediate) {
+    let timeout;
+    return function(e) {
+        const context = e,
+            args = arguments;
+        const later = () => {
+            timeout = null;
+            if (!immediate) func.apply(context, args);
+        };
+        const callNow = immediate && !timeout;
+        clearTimeout(timeout);
+        timeout = setTimeout(later, wait);
+        if (callNow) func.apply(context, args);
+    };
+}
+
+function filterResults(e) {
+    // render entire list from indexDB if input is empty
+    const query = e.target.value
+    if (query === '') {
+        render()
+        return
     }
+    window.indexedDB.open("odinochka", 5).onsuccess = function(event){
+        let db = event.target.result;
+
+        let tx = db.transaction('tabgroups', 'readonly');
+        let store = tx.objectStore('tabgroups');
+
+        let allRecords = store.getAll()
+        allRecords.onsuccess = function() {
+            const filteredResults = allRecords.result.filter(function(cursor) {
+                return cursor.tabs.some(function(tab) {
+                    if (tab.title.toLowerCase().match(query) || tab.url.toLowerCase().match(query)) {
+                        return true
+                    }
+                    return false
+                })
+            })
+            render(filteredResults)
```

Comment:
@nfultz this needs to be tweaked a bit - right now it's returning the whole group if any of its children match the filter query.  It should filter out unmatched children as well.